vsphere/datadog_checks/vsphere/vsphere.py

```diff
@@ -922,26 +905,25 @@ def check(self, instance):
 
             # Second part: do the job
             self.collect_metrics(instance)
-        else:
-            self.log.debug("Thread pool is still processing jobs from previous run. Not scheduling anything.")
 
-        self._query_event(instance)
+            self._query_event(instance)
 
-        thread_crashed = False
-        try:
-            while True:
-                self.log.error(self.exceptionq.get_nowait())
-                thread_crashed = True
-        except Empty:
-            pass
+            try:
+                thread_crashed = False
+                while True:
+                    # Pop every exceptions from the queue and log them
+                    self.log.error(self.exceptionq.get_nowait())
```

Comment:
as discussed offline this logic needs to be changed: this now runs only once, right after the pool was started, so there's a good chance almost no jobs were run and that no crashes will be logged even though a lot of jobs that haven't run yet may fail.

Fix: log exceptions as soon as they happen, but don't bother terminating the entire thread pool when one job fails